src/main/java/uk/gov/register/util/HashValue.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.register.util;
+
+import uk.gov.register.core.HashingAlgorithm;
+
+public class HashValue {
+    private final String value;
+    private final String hashingAlgorithm;
+
+    public HashValue(HashingAlgorithm hashingAlgorithm, String value) {
+        this.hashingAlgorithm = hashingAlgorithm.toString();
+        this.value = value;
+    }
+
+    public String encode() {
+        return hashingAlgorithm + ":" + value;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public static HashValue decode(String hashingAlgorithm, String encodedHash) {
+        if (!encodedHash.startsWith(hashingAlgorithm)) {
+            throw new RuntimeException("Hash cannot be decoded");
+        }
+
+        String[] parts = encodedHash.split(hashingAlgorithm + ":");
+
+        if (parts.length != 2) {
+            throw new RuntimeException(String.format("Cannot create HashValue from encoded hash %s", encodedHash));
+        }
+
```

Comment:
`sha-256abcde:fghijk` would be picked up by the fact that `encodedHash.split(hashingAlgorithm + ":")` would not be able to split the string.

Point taken on checked exceptions.